src/App.tsx

```diff
@@ -1,26 +1,192 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useState,
+  useCallback,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getFilteredTodos } from './utils/getFilteredTodos';
+
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotifications';
+import { TodoItemTemp } from './components/TodoItemTemp';
+
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { FilterStatus } from './types/FilterStatus';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [errorMessage, setErrorMessage] = useState(Errors.DEFAULT);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.ALL);
+
+  useEffect(() => {
+    setErrorMessage(Errors.DEFAULT);
+    setIsLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(Errors.LOADING_TODOS))
+      .finally(() => setIsLoading(false));
+  }, []);
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.